visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts

```diff
@@ -42,24 +44,34 @@ export class FileChooserController {
 					} else {
 						content = (<any>event.target).result
 					}
+				}
+				reader.onloadend = () => {
+					readFiles++
+					this.addNameDataPair(file.name, content)
```

Comment:
Nit (only address if you feel like it): I would inline this function. If you want to keep the function small, I would just move the whole `onloadend` function out as helper function below the imports: `reader.onloadend = onloadend.bind(this)`